src/dynamic_sharing_service/policy/dynamic.policy.builder.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+
+const DynamicPolicy = require('./dynamic.policy');
+const WantedAttributeBuilder = require('./wanted.attribute.builder');
+const constants = require('../../yoti_common/constants');
+
+const SELFIE_AUTH_TYPE = 1;
+const PIN_AUTH_TYPE = 2;
+
+module.exports = class DynamicPolicyBuilder {
+  constructor() {
+    this.wantedAttributes = {};
+    this.wantedAuthTypes = [];
+  }
+
+  /**
```

Comment:
not sue the comments add value here it would be better to add more detail about what the function is doing rather than just repeating what is in thf function params

i.e.
wantedAttribute - str, this is the wanted attribute object that is used to get the attribute by object key

Reply:
this you could also use a tool like js docs to allow users to get generated documentation

Reply:
We are declaring the expected type to be passed, which should be helpful in generated docs, but I agree, it would be useful to give descriptions for the methods that aren't so obvious.

Reply:
Discussed this briefly with @echarrod - I've now documented the classes in 75f40bb so that it's documented in one place (rather than each method that it's passed to)

Reply:
👍